### ISSUE TITLE
Invalid Authentication Key

### ISSUE THREAD

**Issue by danfratamico** [battlesnake/ddns#2]
Hello,

I am hosting your script and web interface at http://dyn.pulsedns.com; however, I am getting the "invalid authentication key" error. I have inputted my own password in the index.php file. Can you provide steps to fix this and/or more detailed installation instructions? Thanks!

Dan

**Comment by battlesnake** [comment]
[Relevant line of code for password check](https://github.com/battlesnake/ddns/blob/6f70d193424fdc568d10205610b395cda1cd2b46/var/www/public_html/index.php#L106)

[Relevant line of code for setting password](https://github.com/battlesnake/ddns/blob/6f70d193424fdc568d10205610b395cda1cd2b46/var/www/public_html/index.php#L4)

Ensure the password you set matches the one you're sending.  Also note that passwords are sent in plaintext (not encrypted), so it would be advisable to set up HTTPS for your site (can be done for free with LetsEncrypt).

Use debug logging to test what value password is received in the request vs. what value is being tested against.  Also check the data types.  File a bug report if you find that this is caused by a problem with ddns itself.

**Comment by danfratamico** [comment]
Ok, thank you. I did already set the password and connected my database using my powerdns credentials. Either way, the index.php is still displaying the invalid key error. Also, I will be using DDNS with DD-WRT, see screenshot. Do you know what I should enter for the following?
<img width="591" alt="screen shot 2017-08-21 at 1 12 34 pm" src="https://user-images.githubusercontent.com/3915913/29530610-94b0fb6e-8672-11e7-95ad-e75b93ba1979.png">

**Comment by battlesnake** [comment]
DD-WRT's DDNS isn't compatible with my ddns, as far as I know.  You can probably set up a cronjob on the router instead, which `curl`s the ddns server periodically instead.

**Comment by danfratamico** [comment]
Yes, I can set that up. The part I am confused with is passing the password to my http://dyn.pulsedns.com/index.php URL (is there no username I should be passing?). How would I go about that? Also, would you be able to provide an example of the CRON entry?

**Comment by battlesnake** [comment]
If I recall correctly, it's passed as a query parameter in the URL.

**Comment by danfratamico** [comment]
I believe I got past this error, but now I am unable to UPDATE. I am using the web GUI and entered the following A record "subdomain.domain.com" but I am getting a "Parse error: invalid name". I'm thinking this has to do with the DOMAIN= setting in the SH file. What does that need to be set to?

**Comment by danfratamico** [comment]
Hi Mark, any idea about my question above?

**Comment by battlesnake** [comment]
I made this project several years ago as a quick hack - I don't remember how it was supposed to work so you'd be best checking the source.
